File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        closeToOpen = { ")": "(", "}": "{", "]": "["}

        for c in s:
            if c in closeToOpen.values():
                stack.append(c)
            elif c in closeToOpen.keys():
                if len(stack) == 0:
                    return False
                elif stack[-1] == closeToOpen[c]:
                    stack.pop()
                else:
                    return False        
        return len(stack) == 0

File: Data Structures & Algorithms/minimum-stack/submission-2.cs
public class MinStack {

	// Val, minimium before pushing
	Stack<(int, int)> stack;
	int currMinimum = int.MaxValue;

    public MinStack() {
    	stack = new Stack<(int, int)>();
    }
    
    public void Push(int val) {
        stack.Push((val, currMinimum));
		currMinimum = Math.Min(currMinimum, val);
    }
    
    public void Pop() {
        (int value, int val) = stack.Pop();
		// Set minimum to what is was before
		currMinimum = val;
    }
    
    public int Top() {
        return stack.Peek().Item1;
    }
    
    public int GetMin() {
		return currMinimum;
    }
}
